packages/frontend-shared/src/gql-components/Auth.vue

```diff
@@ -64,53 +86,69 @@ fragment Auth on Query {
     email
     fullName
   }
-  isAuthBrowserOpened
+  authState {
+    browserOpened
+    name
+    message
+  }
 }
 `
 
 gql`
-mutation Logout {
+mutation Auth_Logout {
   logout {
     ...Auth
   }
 }
 `
 
 gql`
-mutation Login {
+mutation Auth_Login {
   login {
     ...Auth
   }
 }
 `
 
 gql`
-query BrowserOpened {
-  isAuthBrowserOpened
+mutation Auth_ResetAuthState {
+  resetAuthState {
+    ...Auth
+  }
 }
 `
 
-const login = useMutation(LoginDocument)
-const logout = useMutation(LogoutDocument)
+gql`
+query Auth_BrowserOpened {
+  authState {
+    browserOpened
+    name
+    message
+  }
+}
+`
+
+const login = useMutation(Auth_LoginDocument)
+const logout = useMutation(Auth_LogoutDocument)
+const reset = useMutation(Auth_ResetAuthStateDocument)
 const loginButtonRef = ref(Button)
 
 onMounted(() => {
   loginButtonRef?.value?.$el?.focus()
 })
 
-const error = ref<string>()
 const clickedOnce = ref(false)
 
 const emit = defineEmits<{
   (event: 'continue', value: boolean): void
 }>()
 
 const viewer = computed(() => props.gql.cloudViewer)
-const isBrowserOpened = computed(() => props.gql.isAuthBrowserOpened)
+const isBrowserOpened = computed(() => props.gql.authState.browserOpened)
```

Comment:
I know you didn't introduce this code here, but what is the point of this computed property, other than adding more reactive effects to track? We could just replace `isBrowserOpened.value` with `props.gql.authState.browserOpened` in this file and remove the need for the temp variable. What do you think?

Reply:
I think the only point is making it is easier to read later when the value is used. `props.gql.authState.browserOpened` is a bit harder to parse at a glance. In this file when we've pulled out `viewer` and `isBrowserOpened` then the conditional logic that depends on them reads a bit more naturally to me, example:

```js
const buttonMessage = computed(() => {
  if (!isBrowserOpened.value && isLoggingIn.value) {
    return t('topNav.login.actionOpening')
  }

  if (!clickedOnce.value && !viewer.value) {
    return t('topNav.login.actionLogin')
  }

  if (isLoggingIn.value) {
    return t('topNav.login.actionWaiting')
  }

  if (viewer.value) {
    return t('topNav.login.actionContinue')
  }

  // default
  return t('topNav.login.actionLogin')
})
```

👆 Some of those are gql-driven, some are not. 

But if the extra computed values are a problem for performance, or we just want to standardize on not doing this, that's fine. I don't think this is a huge deal, just more readable.